aspnetcore/signalr/diagnostics.md

```diff
@@ -0,0 +1,206 @@
+
+| Level | Description |
+| ----- | ----------- |
+| `None` | No messages are logged. |
+| `Critical` | Messages that indicate a failure in the entire app. |
+| `Error` | Messages that indicate a failure in the current operation. |
+| `Warning` | Messages that indicate a non-fatal problem. |
+| `Information` | Informational messages. |
+| `Debug` | Diagnostic messages useful for debugging. |
+| `Trace` | Very detailed diagnostic messages designed for diagnosing specific issues. |
+
+Once you've configured the verbosity, the logs will be written to the Browser Console (or Standard Output in a NodeJS application).
+
+If you want to send logs to a custom logging system, you can provide a JavaScript object implementing the `ILogger` interface. The only method that needs to be implemented is `log`, which takes the level of the event and the message associated with the event. For example:
+
+[!code-typescript[Creating a custom logger](diagnostics/custom-logger.ts?highlight=3-7,13)]
+
+## .NET client logging
+
+> [!NOTE]
+> Client-side logs may contain sensitive information from your application. **Never** post raw logs from production applications to public forums like GitHub.
+
+To get logs from the .NET client, you can use the `.ConfigureLogging` method on `HubConnectionBuilder`. This works the same way as the `.ConfigureLogging` method on `WebHostBuilder` and `HostBuilder`. You can configure the same logging providers you use in ASP.NET Core. However, you have to manually install and enable the NuGet packages for the individual logging providers.
+
+### Console logging
+
+In order to enable Console logging, add the `Microsoft.Extensions.Logging.Console` nuget package. Then, use the `AddConsole` method to configure the console logger:
+
+[!code-csharp[Configuring console logging in .NET client](diagnostics/net-client-console-log.cs?highlight=6)]
+
+### Debug output window logging
+
+You can also configure logs to go to the Output Window in Visual Studio. Install the `Microsoft.Extensions.Logging.Debug` package and use the `AddDebug` method:
+
+[!code-csharp[Configuring debug output window logging in .NET client](diagnostics/net-client-debug-log.cs?highlight=6)]
+
+### Other logging providers
+
+You can also use other logging providers such as Serilog, Seq, NLog, or any other logging system that integrates with `Microsoft.Extensions.Logging`. If your logging system provides an `ILoggerProvider`, you can register it with `.AddProvider`:
+
+[!code-csharp[Configuring a custom logging provider in .NET client](diagnostics/net-client-custom-log.cs?highlight=6)]
+
+### Controlling verbosity
+
+If the debug logs are too verbose, you can set the default minimum level to `Information` and just increase the logging level for SignalR logs. This can be done in code, in much the same was as on the server:
```

Comment:
Nit:

```suggestion
If the debug logs are too verbose, you can set the default minimum level to `Information` to increase the logging level for SignalR logs. This can be done in code, in much the same was as on the server:
```

Reply:
That's not quite accurate. What I'm describing is **lowering** the default verbosity and then **increasing** the verbosity just for SignalR. It definitely feels clunky though.

Reply:
Got an attempt to de-clunk in the next iteration.